api/src/services/agencies_v1/get_agencies.py

```diff
@@ -17,13 +19,47 @@
 class AgencyFilters(BaseModel):
     agency_id: int | None = None
     agency_name: str | None = None
+    active: bool | None = None
 
 
 class AgencyListParams(BaseModel):
     pagination: PaginationParams
 
     filters: AgencyFilters | None = Field(default_factory=AgencyFilters)
 
+def _construct_active_query(stmt: Select) -> Select:
+    active_agency = (
+        select(Agency.agency_id)
+        .join(Opportunity, onclause=Agency.agency_code == Opportunity.agency_code)
+        .join(CurrentOpportunitySummary)
+        .where(Agency.is_test_agency.isnot(True))  # Exclude test agencies
+        .where(
+            CurrentOpportunitySummary.opportunity_status.in_(
+                [OpportunityStatus.FORECASTED, OpportunityStatus.POSTED]
+            )
+        )
+    )
+
+    active_top_level_agency = (
+        (
+            select(Agency.top_level_agency_id)
+            .join(Opportunity, onclause=Agency.agency_code == Opportunity.agency_code)
+            .join(CurrentOpportunitySummary)
+            .where(Agency.is_test_agency.isnot(True))  # Exclude test agencies
+            .where(
+                CurrentOpportunitySummary.opportunity_status.in_(
+                    [OpportunityStatus.FORECASTED, OpportunityStatus.POSTED]
+                )
+            )
+        )
+        .union(active_agency)
+        .subquery()
+    )
+
+    agency_id_stmt = select(active_top_level_agency).distinct
```

Comment:
Wouldn't this need to be `.distinct()` - does it work if you don't call the function?